### PR TITLE
docs: sd-notification docs [skip chromatic]

### PR DESCRIPTION
**Pull request by smfonseca**
## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/components/notification/notification.stories.ts [hunk 1]
```diff
@@ -3,21 +3,27 @@
...
 const { overrideArgs } = storybookHelpers('sd-notification');
 
+/**
+ * Used to communicate important information or status to the user. Notifications can be displayed inline or as a toast / toast stack.
+ *
+ *  **Related templates**:
+ * - [Toast Notification](?path=/docs/templates-toast-notification--docs)
+ */
+
 export default {
```
[the remark this answers is not in the document]
Comment (coraliefeil) [reply packages/components/src/components/notification/notification.stories.ts:18]: Yes, pls show default slot in the intro.

File: packages/components/src/templates/toast-notification.stories.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import '../solid-components';
+import { html } from 'lit';
+import { userEvent } from '@storybook/test';
+
+export default {
+  title: 'Templates/Toast Notification',
```
Comment (coraliefeil) [inline packages/components/src/templates/toast-notification.stories.ts:8]: Why is this a template? I would see this in the docs as we miss the toasts ATM.

**Comment by coraliefeil** [comment]
Refined:
[Figma Docs](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=2265-6174&t=qkAaIdEH5Fp0KY14-11)
[Figma Template](https://www.figma.com/design/YDktJcseQIIQbsuCpoKS4V/Component-Docs?node-id=2951-15059&t=qkAaIdEH5Fp0KY14-11)

File: packages/components/src/components/notification/notification.stories.ts [hunk 3]
```diff
@@ -56,295 +63,81 @@ export const Default = {
...
-      args,
-      constants: [
-        { type: 'attribute', name: 'duration', value: 10000 },
-        { type: 'attribute', name: 'open', value: true }
-      ]
-    });
-  }
+  name: 'Variants',
+  render: () => html`
+    <sd-notification variant="info" open>Lorem ipsum dolor sit</sd-notification>
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/components/src/components/notification/notification.stories.ts:73]: Updated

File: packages/components/src/components/notification/notification.stories.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/components/src/components/notification/notification.stories.ts:18]: Not sure I understood the request. As far as I can see it's the same. Can you clarify?

File: packages/components/src/templates/toast-notification.stories.ts (hunk 2 above)
Comment (smfonseca) [reply packages/components/src/templates/toast-notification.stories.ts]: Moved into the docs.

**Comment by smfonseca** [comment]
@coraliefeil I've created the new templates from figma. Please have a look

File: packages/components/src/components/notification/notification.stories.ts [hunk 4]
```diff
@@ -53,85 +60,98 @@ export const Default = {
...
-      args,
-      constants: { type: 'attribute', name: 'open', value: true }
-    });
-  }
+  name: 'Variants',
+  render: () => html`
+    <sd-notification variant="info" open>Lorem ipsum dolor sit</sd-notification>
+    <sd-notification variant="success" open>Lorem ipsum dolor sit</sd-notification>
+    <sd-notification variant="error" open>Lorem ipsum dolor sit</sd-notification>
+    <sd-notification variant="warning" open>Lorem ipsum dolor sit</sd-notification>
```
Comment (yoezlem) [inline packages/components/src/components/notification/notification.stories.ts:76]:
```suggestion
    <sd-notification variant="info" open>Info Lorem ipsum dolor sit</sd-notification>
    <sd-notification variant="success" open>Success Lorem ipsum dolor sit</sd-notification>
    <sd-notification variant="error" open>Error Lorem ipsum dolor sit</sd-notification>
    <sd-notification variant="warning" open>Warning Lorem ipsum dolor sit</sd-notification>
```

**Comment by yoezlem** [comment]
Rest LGTM

File: packages/components/src/components/notification/notification.stories.ts [hunk 5]
```diff
@@ -3,21 +3,27 @@
...
 
 const { argTypes, parameters } = storybookDefaults('sd-notification');
 const { generateTemplate } = storybookTemplate('sd-notification');
 const { overrideArgs } = storybookHelpers('sd-notification');
 
+/**
+ * Used to communicate important information or status to the user. Notifications can be displayed inline or as a toast / toast stack.
+ *
+ *  **Related templates**:
+ * - [Notifications](?path=/docs/templates-notifications--docs)
```
Comment (yoezlem) [inline packages/components/src/components/notification/notification.stories.ts:15]: In Figma we have "Notification" here "Notifications".
Please align with Design

**Comment by yoezlem** [comment]
Rest LGTM

File: packages/components/src/components/notification/notification.stories.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (coraliefeil) [reply packages/components/src/components/notification/notification.stories.ts:18]: Now it’s good:

image
Bildschirmfoto 2024-08-29 um 21 24 28
THX

File: packages/components/src/components/notification/notification.stories.ts [hunk 6]
```diff
@@ -53,85 +60,98 @@ export const Default = {
...
-      axis: {
-        y: { type: 'attribute', name: 'variant' }
-      },
-      args,
-      constants: { type: 'attribute', name: 'open', value: true }
-    });
-  }
+  name: 'Variants',
+  render: () => html`
+    <sd-notification variant="info" open>Lorem ipsum dolor sit</sd-notification>
```
Comment (coraliefeil) [inline packages/components/src/components/notification/notification.stories.ts:73]: <sd-notification variant="info" open>Info Lorem ipsum dolor sit</sd-notification>

File: packages/components/src/components/notification/notification.stories.ts [hunk 7]
```diff
@@ -53,85 +60,98 @@ export const Default = {
...
+    <sd-notification variant="success" open>Lorem ipsum dolor sit</sd-notification>
```
Comment (coraliefeil) [inline packages/components/src/components/notification/notification.stories.ts:74]: <sd-notification variant="success" open>Success Lorem ipsum dolor sit</sd-notification>

File: packages/components/src/components/notification/notification.stories.ts [hunk 8]
```diff
@@ -53,85 +60,98 @@ export const Default = {
...
+    <sd-notification variant="error" open>Lorem ipsum dolor sit</sd-notification>
```
Comment (coraliefeil) [inline packages/components/src/components/notification/notification.stories.ts:75]: <sd-notification variant="error" open>Error Lorem ipsum dolor sit</sd-notification>

File: packages/components/src/templates/notifications.stories.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+export default {
+  title: 'Templates/Notifications',
+  tags: ['!dev'],
+  parameters: {
+    chromatic: { disableSnapshot: true }
+  }
+};
+
+export const InfoNotification = {
```
Comment (coraliefeil) [inline packages/components/src/templates/notifications.stories.ts:14]: Set padding-right to 16px 
Bildschirmfoto 2024-08-29 um 21 30 57

File: packages/components/src/templates/notifications.stories.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+      <sd-notification class="info-variant" variant="info" open>
+        Editing is restricted
+        <sd-button variant="secondary">Request to edit</sd-button>
+      </sd-notification>
+    `;
+  }
+};
+
+export const InfoNotificationInline = {
+  name: 'Info Notification Inline',
```
Comment (coraliefeil) [inline packages/components/src/templates/notifications.stories.ts:35]: name: 'Info Toast Notification',
  
  Is it possible to show the shadow?
  
  I wanted to have at least 1 toast as a template people can grab ...

File: packages/components/src/templates/notifications.stories.ts [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    `;
+  }
+};
+
+export const ErrorNotification = {
+  name: 'Error  Notification',
+  render: (_args: Record<string, any>) => {
+    return html`
+      <sd-notification variant="error" open closable>
+        <p class="sd-paragraph">A problem occurred while submitting your data. <sd-link>More information</sd-link></p>
```
Comment (coraliefeil) [inline packages/components/src/templates/notifications.stories.ts:77]: sd-link color should be text.primary

File: packages/components/src/components/notification/notification.stories.ts (hunk 5 above)
Comment (smfonseca) [reply packages/components/src/components/notification/notification.stories.ts]: Updated on storybook

File: packages/components/src/templates/notifications.stories.ts (hunk 9 above)
Comment (smfonseca) [reply packages/components/src/templates/notifications.stories.ts]: Updated

File: packages/components/src/templates/notifications.stories.ts (hunk 10 above)
Comment (smfonseca) [reply packages/components/src/templates/notifications.stories.ts]: Added toast as discussed in sync call

File: packages/components/src/templates/notifications.stories.ts (hunk 11 above)
Comment (smfonseca) [reply packages/components/src/templates/notifications.stories.ts]: Updated

File: packages/components/src/templates/notifications.stories.ts [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    return html`
+      <style>
+        sd-notification.inline-variant::part(content) {
+          width: auto;
+        }
+      </style>
+
+      <sd-notification class="inline-variant" variant="info" open closable>
+        Event deleted
+        <sd-button class="ml-4" variant="secondary">Undo</sd-button>
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply packages/components/src/templates/notifications.stories.ts]: Updated

File: packages/components/src/templates/notification.stories.ts [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+      <sd-notification class="info-variant" variant="info" open>
+        Editing is restricted
+        <sd-button variant="secondary" class="mr-4">Request to edit</sd-button>
+      </sd-notification>
+    `;
+  }
+};
+
+export const InfoToastNotification = {
```
Comment (coraliefeil) [inline packages/components/src/templates/notification.stories.ts:34]: That’s not working inside the iframe I guess ... and the padding left & right for button/secondary are incorrect ...

**Comment by coraliefeil** [comment]
One last thing ... see comment

**Review by yoezlem — CHANGES_REQUESTED** [review]
In Design we have more space between the notifications
Figma:
image

Storybook:
image

**Comment by yoezlem** [comment]
Rest LGTM

File: packages/components/src/templates/notification.stories.ts (hunk 13 above)
Comment (coraliefeil) [reply packages/components/src/templates/notification.stories.ts:34]: Now the toast pops up. THX

But:

padding left & right for button/secondary are incorrect:
Bildschirmfoto 2024-09-03 um 12 53 09
Should be 16px

Comment (smfonseca) [reply packages/components/src/templates/notification.stories.ts:34]: Updated

**Comment by smfonseca** [comment]
> In Design we have more space between the notifications Figma: image
> 
> Storybook: image

@yoezlem margin updated

File: packages/components/src/components/notification/notification.stories.ts [hunk 14]
```diff
@@ -53,85 +60,105 @@ export const Default = {
...
+        setTimeout(() => {
+          durationIndicator.open = true;
+        }, 3000);
+      });
+    </script>
+  `
 };
 
 /**
  * Display a toast notification at the top-right of the screen by using the `toast` method. Click on the **Show code** button to see the JavaScript code responsible for generating the toast notification. The default position is `top-right`.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/notification/notification.stories.ts:158]: For the preview can you do something similar to what we have in FIGMA?

File: packages/components/src/components/notification/notification.stories.ts [hunk 15]
```diff
@@ -189,22 +216,13 @@ export const ToastNotification = {
...
     `;
-  },
-  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
-    const button = canvasElement.querySelector('#top-right');
-    await userEvent.click(button!);
   }
 };
 
 /**
  * Display a toast notification positioned at the bottom-center of the screen by setting the `toastStack` attribute to `bottom-center`. Click on the **Show code** button to see the JavaScript code responsible for generating the toast notification.
```
Comment (MartaPintoTeixeira) [inline packages/components/src/components/notification/notification.stories.ts:223]: For the preview can you do something similar to what we have in FIGMA?

Comment (MartaPintoTeixeira) [inline packages/components/src/components/notification/notification.stories.ts:223]: In figma de have both bottom and top notifications under "Toast Notification". Is this possible in CODE?

**Comment by coraliefeil** [comment]
Approved. THX

**Review by yoezlem — APPROVED** [review]

**Comment by Vahid1919** [comment]
@smfonseca 
Just a reminder regarding the build issues here as well, I have already merged my changes that included the fix, so merging docs/re-write-docs might be an easy fix.

**Pull request merged**